### PR TITLE
docs: windows: optimize line endings

### LINKED ISSUE #5114: docs: windows: add mitigation for already existing CRLF
### Report

https://dvc.org/doc/user-guide/how-to/run-dvc-on-windows#how-to-run-dvc-on-windows

we could also suggest using .gitattributes and eol to automatically convert line endings to lf.

### PR DESCRIPTION
**Pull request by aayushk9**
Fixes #5114 

Enhanced DVC compatibility on Windows by setting LF line endings using .gitattributes and set core.autocrlf to false whereas core.eol to lf.

### CODE REVIEW

**Comment by shcheklein** [comment]
hey @aayushk9 - was it closed intentionally?

**Comment by aayushk9** [comment]
@shcheklein Apologies for any confusion. I noticed that some checks are failing for the pull request. I've reopened it, if there's anything specific that needs attention please let me know.

File: content/docs/user-guide/how-to/.pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
```
Comment (shcheklein) [inline content/docs/user-guide/how-to/.pre-commit-config.yaml:3]: this file seems unrelated?

File: content/docs/user-guide/how-to/run-dvc-on-windows.md [hunk 2]
```diff
@@ -73,6 +73,39 @@ For VS Code, add
 
 to your global `settings.json` or to your project's `.vscode/settings.json`.
 
+### Set up LF Line Endings with .gitattributes for DVC on Windows
+
+To enhance DVC compatibility on Windows, it is advisable to employ a
+.gitattributes file with the eol attribute to configure line endings.
```
Comment (shcheklein) [inline content/docs/user-guide/how-to/run-dvc-on-windows.md:79]: .gitattributes -> `.gitattributes` (put into single backquotes to make it like an `inline code`)

File: content/docs/user-guide/how-to/run-dvc-on-windows.md [hunk 3]
```diff
@@ -73,6 +73,39 @@ For VS Code, add
...
+
+Add the following line to your .gitattributes:
```
Comment (shcheklein) [inline content/docs/user-guide/how-to/run-dvc-on-windows.md:81]: `.gitattributes` (inline code)

File: content/docs/user-guide/how-to/run-dvc-on-windows.md [hunk 4]
```diff
@@ -73,6 +73,39 @@ For VS Code, add
...
+```
+{
+
+  git config --global core.autocrlf false
+  git config --global core.eol lf
+
+}
+```
+
+Now git will handle line endings consistently.
```
Comment (shcheklein) [inline content/docs/user-guide/how-to/run-dvc-on-windows.md:107]: Git

File: content/docs/user-guide/how-to/run-dvc-on-windows.md [hunk 5]
```diff
@@ -73,6 +73,39 @@ For VS Code, add
...
 
+### Set up LF Line Endings with .gitattributes for DVC on Windows
+
+To enhance DVC compatibility on Windows, it is advisable to employ a
+.gitattributes file with the eol attribute to configure line endings.
+
+Add the following line to your .gitattributes:
+
+```
+{
```
Comment (shcheklein) [inline content/docs/user-guide/how-to/run-dvc-on-windows.md:84]: we don't need curly braces, right?

File: content/docs/user-guide/how-to/run-dvc-on-windows.md [hunk 6]
```diff
@@ -73,6 +73,39 @@ For VS Code, add
...
+```
+
+This configuration tells Git to treat all files as text files and use LF line
+endings regardless of the platform.
+
+### Configure Git for LF Line Endings
+
+Set core.autocrlf to false and core.eol to lf
+
+```
```
Comment (shcheklein) [inline content/docs/user-guide/how-to/run-dvc-on-windows.md:98]: use  "```cli"

use $ before each CLI command:

```cli
$ git config --global core.autocrlf false
$ git config --global core.eol lf
```

**Review by shcheklein — COMMENTED** [review]
Good stuff, thanks! Some style changes and updates, otherwise looks good.

File: content/docs/user-guide/how-to/.pre-commit-config.yaml (hunk 1 above)
Comment (aayushk9) [reply content/docs/user-guide/how-to/.pre-commit-config.yaml]: I've removed this file

**Review by shcheklein — APPROVED** [review]

**Comment by shcheklein** [comment]
thanks @aayushk9 !
